synapse/rest/client/v2_alpha/auth.py

```diff
@@ -95,29 +95,32 @@ async def on_POST(self, request, stagetype):
 
             authdict = {"response": response, "session": session}
 
-            success = await self.auth_handler.add_oob_auth(
-                LoginType.RECAPTCHA, authdict, request.getClientIP()
-            )
-
-            if success:
-                html = self.success_template.render()
-            else:
+            try:
+                await self.auth_handler.add_oob_auth(
+                    LoginType.RECAPTCHA, authdict, request.getClientIP()
+                )
+            except LoginError as e:
+                # Authentication failed, let user try again
                 html = self.recaptcha_template.render(
                     session=session,
                     myurl="%s/r0/auth/%s/fallback/web"
                     % (CLIENT_API_PREFIX, LoginType.RECAPTCHA),
                     sitekey=self.hs.config.recaptcha_public_key,
+                    error=e.msg,
```

Comment:
Are we sure that these error messages are suitable for displaying to the user? Looking through the checkers I see a few that raise `LoginError` with a blank message.

Reply:
Yeah you're right, the errors are more for devs and people troubleshooting than users I think. From what I can tell the errors would be caused by a broken client rather than the user.

Reply:
I think we should at least give some string back for those cases so that people can ask a server admin about it. Right now some of them will end up just saying `Error: `.

Reply:
I added a message to the couple LoginErrors that were missing one https://github.com/matrix-org/synapse/pull/10561/commits/8d6a469a43e276a40862eb8a1f2993717572320e